### PR TITLE
LibWeb: Support Google Lens :^)

### PR DESCRIPTION
**Pull request by trflynn89**
Drag-and-drop part 2, electric boogaloo

Google Lens has some interesting interactions with the drag-and-drop interface. The basic flow seems to be:
1. User drags an image onto the upload form
2. Google grabs the file object from the `DragEvent.dataTransfer.items` slot
3. Google creates a `ClipboardEvent` (which also has a `DataTransfer` slot) and attaches the file to that event
4. Some time later, Google takes the `DataTransfer.files` object from the clipboard event and repoints the upload form's `HTMLInputElement.selectedFiles` at this files list
5. The form is submitted to lens.google.com via POST

So this PR implements the IDL interfaces involved here, and fixes a couple of form submission bugs. Those bugs don't include tests here, unfortunately - the code paths here are only hit for HTTPS POSTs.

https://github.com/user-attachments/assets/8be3497f-f7b2-4f12-8ce7-865b40d3668b

* Requires spoofing the UA 😢

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/DataTransferItemList.cpp [hunk 1]
```diff
@@ -108,4 +110,16 @@ JS::GCPtr<DataTransferItem> DataTransferItemList::add(JS::NonnullGCPtr<FileAPI::
...
 }
 
+// https://html.spec.whatwg.org/multipage/dnd.html#dom-datatransferitemlist-item
+Optional<JS::Value> DataTransferItemList::item_value(size_t index) const
+{
+    // To determine the value of an indexed property i of a DataTransferItemList object, the user agent must return a
+    // DataTransferItem object representing the ith item in the drag data store. The same object must be returned each
+    // time a particular item is obtained from this DataTransferItemList object. The DataTransferItem object must be
+    // associated with the same DataTransfer object as the DataTransferItemList object when it is first created.
+    if (index <= m_data_transfer->length())
```
Comment (shannonbooth) [inline Userland/Libraries/LibWeb/HTML/DataTransferItemList.cpp:120]: `if (index < m_data_transfer->length())`?

Comment (trflynn89) [reply Userland/Libraries/LibWeb/HTML/DataTransferItemList.cpp]: :man_facepalming:  wow, thanks! Fixed, and included the edge case in the test

**Comment by shannonbooth** [comment]
Looks good to me!
